# Extracting Information from an Exception
from sys import exception


# Write a function read_integer that takes a string and tries to convert it to an integer.
# If a ValueError occurs, handle the exception and print the error arguments and the error type.
# Additionally, save the exception in a variable and print it outside the except block.

### üá∫üá¶ Ukrainian version:

# –í–∏—Ç—è–≥ —ñ–Ω—Ñ–æ—Ä–º–∞—Ü—ñ—ó –∑ –≤–∏–Ω—è—Ç–∫—É

# –ù–∞–ø–∏—à—ñ—Ç—å —Ñ—É–Ω–∫—Ü—ñ—é read_integer, —è–∫–∞ –ø—Ä–∏–π–º–∞—î —Ä—è–¥–æ–∫ —ñ –Ω–∞–º–∞–≥–∞—î—Ç—å—Å—è –ø–µ—Ä–µ—Ç–≤–æ—Ä–∏—Ç–∏ –π–æ–≥–æ –Ω–∞ —Ü—ñ–ª–µ —á–∏—Å–ª–æ.
# –Ø–∫—â–æ –≤–∏–Ω–∏–∫–∞—î ValueError, –æ–±—Ä–æ–±—ñ—Ç—å –≤–∏–Ω—è—Ç–æ–∫ —Ç–∞ –≤–∏–≤–µ–¥—ñ—Ç—å –∞—Ä–≥—É–º–µ–Ω—Ç–∏ –ø–æ–º–∏–ª–∫–∏ —ñ —Ç–∏–ø –ø–æ–º–∏–ª–∫–∏.
# –î–æ–¥–∞—Ç–∫–æ–≤–æ, –∑–±–µ—Ä–µ–∂—ñ—Ç—å –≤–∏–Ω—è—Ç–æ–∫ —É –∑–º—ñ–Ω–Ω—ñ–π —ñ –≤–∏–≤–µ–¥—ñ—Ç—å —ó—ó –∑–∞ –º–µ–∂–∞–º–∏ –±–ª–æ–∫–∞ except.

# Write your code here
def read_integer(a):
    num_a = None
    exception = None
    try:
        num_a = int(a)
    except ValueError as e:
        exception = e
        print(f"Arguments of mistakes: {e.args}")
        print(f"Type of exception: {type(e)}")
        return exception
    return num_a or exception

print(read_integer("2"))
print(read_integer("a"))